server/jetstream_cluster.go

```diff
@@ -4271,7 +4278,7 @@ func (cc *jetStreamCluster) selectPeerGroup(r int, cluster string, cfg *StreamCo
 			}
 		}
 		// Add to our list of potential nodes.
-		nodes = append(nodes, wn{p.ID, available})
+		nodes = append(nodes, wn{p.ID, available, ha})
```

Comment:
Ha won't cover r16 and includes consumer

Reply:
Not following the R16 comment.. But yes, if we are placing a replicated stream I think we try to balance on HA assets per peer over available memory. Note that for the time being, all nodes can handle the stream from a storage perspective if max bytes was specified.

Reply:
What I am not understanding is:
If you have 5 empty R1 streams already placed.
How are they accounted for when placing anything else?

Reply:
Currently they are not formally, but we shuffle the peers to randomize if available is all the same.

Reply:
The test I wrote shows that the test Marco was trying we now randomize placements. If you specific MaxBytes it worked ok already.